File: leetcode_problems/remove_element.py
'''
Input: nums = [0,1,2,2,3,0,4,2], val = 2
Output: 5, nums = [0,1,4,0,3,_,_,_]
Explanation: Your function should return k = 5, with the first five elements of nums containing 0, 0, 1, 3, and 4.
Note that the five elements can be returned in any order.
It does not matter what you leave beyond the returned k (hence they are underscores).
'''


nums=[0,1,2,2,3,0,4,2]
val=2
count=0

for i in range(len(nums)):
    for j in range(i+1,len(nums)):
        if(nums[i]==val):
            nums[j],nums[i]=nums[i],nums[j]


for i in range(len(nums)):
    if val ==nums[i]:
        count+=1
        nums[i]="_"
k=(len(nums))-count

print(nums)

print(k)

File: DS_and_A/Array/removeelement.java
package Array;

/**
 * removeelement
 */
public class removeelement {

    public static void main(String[] args) {
        int [] a = {3,2,2,3,2,2,2,2,2,3,3,3,3};
        int val = 3;
        //output should be [2,2,_,_];

        //first move all the val to the end 
        int i =0;
        int j=a.length-1;
        while(i<=j){
            if(a[i]==val && a[j]!=val){
                int temp=a[i];
                a[i]=a[j];
                a[j]=temp;

                i++;
                j--;

            }
            else if(a[i]==val){
                j--;
            }else{
                i++;
            }
        }
        
        for(int k=0;k<a.length;k++){
            if(a[k]==val){
                a[k]=0;
            }
        }

    }

}